File: src/main/java/cn/cloudself/model/AddressEntity.java
package cn.cloudself.model;

import org.hibernate.annotations.DynamicInsert;

import javax.persistence.*;

/**
 * @author HerbLuo
 * @version 1.0.0.d
 */
@Entity
@DynamicInsert
@Table(name = "address", schema = "shop")
public class AddressEntity {
    private int id;
    private boolean enabled;
    private String address;
    private String postCode;
    private String receiverName;
    private String phone;
    private int userId;
    private boolean areDefault;

    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Basic
    @Column(name = "enabled", nullable = false)
    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    @Basic
    @Column(name = "address", nullable = false, length = 255)
    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    @Basic
    @Column(name = "post_code", nullable = false, length = 6)
    public String getPostCode() {
        return postCode;
    }

    public void setPostCode(String postCode) {
        this.postCode = postCode;
    }

    @Basic
    @Column(name = "receiver_name", nullable = false, length = 12)
    public String getReceiverName() {
        return receiverName;
    }

    public void setReceiverName(String receiverName) {
        this.receiverName = receiverName;
    }

    @Basic
    @Column(name = "phone", nullable = false, length = 16)
    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    @Basic
    @Column(name = "user_id", nullable = false)
    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    @Basic
    @Column(name = "is_default", nullable = false)
    public boolean isAreDefault() {
        return areDefault;
    }

    public void setAreDefault(boolean areDefault) {
        this.areDefault = areDefault;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        AddressEntity that = (AddressEntity) o;

        if (id != that.id) return false;
        if (enabled != that.enabled) return false;
        if (userId != that.userId) return false;
        if (areDefault != that.areDefault) return false;
        if (address != null ? !address.equals(that.address) : that.address != null) return false;
        if (postCode != null ? !postCode.equals(that.postCode) : that.postCode != null) return false;
        if (receiverName != null ? !receiverName.equals(that.receiverName) : that.receiverName != null) return false;
        if (phone != null ? !phone.equals(that.phone) : that.phone != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + (enabled ? 1 : 0);
        result = 31 * result + (address != null ? address.hashCode() : 0);
        result = 31 * result + (postCode != null ? postCode.hashCode() : 0);
        result = 31 * result + (receiverName != null ? receiverName.hashCode() : 0);
        result = 31 * result + (phone != null ? phone.hashCode() : 0);
        result = 31 * result + userId;
        result = 31 * result + (areDefault ? 1 : 0);
        return result;
    }
}

File: src/main/java/cn/cloudself/model/AddressEntity.kt
package cn.cloudself.model

import org.hibernate.annotations.DynamicInsert

import javax.persistence.*

/**
 * @author HerbLuo
 * @version 1.0.0.d
 */
@Entity
@DynamicInsert
@Table(name = "address", schema = "shop")
data class AddressEntity(
        @get:Id
        @get:Column(name = "id", nullable = false)
        @get:GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Int = 0,

        @get:Basic
        @get:Column(name = "enabled", nullable = false)
        var isEnabled: Boolean = false,

        @get:Basic
        @get:Column(name = "address", nullable = false, length = 255)
        var address: String? = null,

        @get:Basic
        @get:Column(name = "post_code", nullable = false, length = 6)
        var postCode: String? = null,

        @get:Basic
        @get:Column(name = "receiver_name", nullable = false, length = 12)
        var receiverName: String? = null,

        @get:Basic
        @get:Column(name = "phone", nullable = false, length = 16)
        var phone: String? = null,

        @get:Basic
        @get:Column(name = "user_id", nullable = false)
        var userId: Int = 0,

        @get:Basic
        @get:Column(name = "is_default", nullable = false)
        var isAreDefault: Boolean = false
)